File: 167. Two Sum II - Input Array is sorted/twoSum.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] result = new int[2];

        int n = numbers.length;
        int index1 = 0;
        int index2 = n - 1;

        while (index1 < n && index2 >= 0) {
            if (numbers[index1] + numbers[index2] == target) {
                return new int[]{index1 + 1, index2 + 1};
            } else if (numbers[index1] + numbers[index2] < target) {
                index1++;
            } else index2--;
        }
        return new int[]{};
    }
}

File: 167. Two Sum II - Input Array is sorted/twoSum.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        result = []

        while left < right:
            sum = numbers[left] + numbers[right]
            if sum == target:
                result.append(left + 1)
                result.append(right + 1)
                break
            elif sum > target:
                right -= 1
            else:
                left += 1
        
        return result